ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -365,47 +366,53 @@ def get_min_stopping_distance(self, ptr_id):
 
         if a > 0.0:
             # currently speeding up
-            time_factor = 0.9
+            time_factor = 0.8
         else:
             time_factor = self.dyn_jmt_time_factor
 
-        # empirical settings to start closer to goal
-        if a < 1.0:
-            decel_rate = 2.0
-        elif a < 2.0:
-            decel_rate = 1.5
-        elif a < 3.0:
-            decel_rate = 1.35
-        elif a < 3.4:
-            decel_rate = 1.2
+        if math.fabs(max_jerk + self.max_desired_jerk) < 0.1:
+            decel_rate = 0.6
         else:
-            decel_rate = 1.0
+            # empirical settings to start closer to goal
+            if a < 1.0:
+                decel_rate = 2.0
+            elif a < 2.0:
+                decel_rate = 1.5
+            elif a < 3.0:
+                decel_rate = 1.35
+            elif a < 3.4:
+                decel_rate = 1.2
+            else:
+                decel_rate = 1.0
 
-        a_dist = get_accel_distance(curpt.JMTD.V, self.min_moving_velocity*1.1, decel_rate,
+        a_dist = get_accel_distance(curpt.JMTD.V, self.min_moving_velocity*1.25, decel_rate,
                                     curpt.JMTD.A)
 
         too_short = False
                                     
         T = get_accel_time(a_dist, curpt.JMTD.V, 0.0) * time_factor
         start = [curpt.JMTD.S, curpt.JMTD.V, curpt.JMTD.A]
-        end = [curpt.JMTD.S + a_dist, self.min_moving_velocity*1.1, self.decel_at_min_moving_velocity]
-        rospy.logdebug("Test decel from v={:3.3f}, a={:3.3f} to v={:3.3f}"
+        end = [curpt.JMTD.S + a_dist, self.min_moving_velocity*1.25, self.decel_at_min_moving_velocity]
+        rospy.logdebug("Test decel_rate {:3.3f} from v={:3.3f}, a={:3.3f} to v={:3.3f}, a={:3.3f} "
                        " in dist {:3.3f} m in time {:3.3f} s"
-                       .format(curpt.JMTD.V, curpt.JMTD.A, self.min_moving_velocity*1.1, a_dist, T))
+                       .format(decel_rate, start[1], start[2], end[1], end[2], a_dist, T))
 
         jmt = JMT(start, end, T)
         jerk = jmt.get_j_at(0.1)
         acc = jmt.get_a_at(T - 0.5)
 
-        rospy.logdebug("found jerk of {:3.2f} with a_dist of {:3.2f}"
-                       " and acc of {:3.2f} with a time of {:3.2f}"
-                       .format(jerk, a_dist, acc ,T))
-        if jerk < 0.0 - self.max_jerk:
+        rospy.logdebug("found initial jerk of {:3.2f}m/s^3 and final acc of {:3.2f}m/s^2"
+                       " when using decel_rate={:3.2f}m/s a_dist ={:3.2f}m and T={:3.2f}s"
+                       .format(jerk, acc, decel_rate, a_dist, T))
+
+        if jerk < max_jerk:
             too_short = True
-            dist_diff = 0.5
+            dist_diff = 1.0
         else:
-            dist_diff = -0.5
-                
+            dist_diff = -1.0
+        if math.fabs(max_jerk + self.max_desired_jerk) < 0.1:
+            dist_diff = 2 * dist_diff
```

Comment:
I doubled the increment distance when using max_desired_jerk since the overall distance will be bigger - maybe it would be better just to use a proportion of the whole length.

Reply:
I'm not sure how it would affect the accuracy vs precision, but setting `dist_diff` as like ~1% of a_dist or something seems reasonable to automatically scale.